js/views/composerview.js

```diff
@@ -394,7 +404,7 @@ define(function(require) {
 					onSuccess();
 				}
 
-				if (this.draftUID !== null) {
+				if (_this.draftUID !== null) {
 					// update UID in message list
 					var collection = Radio.ui.request('messagesview:collection');
 					var message = collection.findWhere({id: this.draftUID});
```

Comment:
Right. When I checked the code I was actually looking at the code responsible for sending new messages, hence I didn't see the issue. Anyway, fixed it in https://github.com/nextcloud/mail/pull/420 :)